scipy/sparse/_base.py

```diff
@@ -872,29 +876,36 @@ def _getcol(self, j):
         # Subclasses should override this method for efficiency.
         # Post-multiply by a (n x 1) column vector 'a' containing all zeros
         # except for a_j = 1
-        n = self.shape[1]
+        N = self.shape[-1]
         if j < 0:
-            j += n
-        if j < 0 or j >= n:
+            j += N
+        if j < 0 or j >= N:
             raise IndexError("index out of bounds")
         col_selector = self._csc_container(([1], [[j], [0]]),
-                                           shape=(n, 1), dtype=self.dtype)
-        return self @ col_selector
+                                           shape=(N, 1), dtype=self.dtype)
+        result = self @ col_selector
+        if self.ndim == 1:
+            # only here to force col to be 2d ("as an (mx1) sparse array")
+            # should this be 1d? -- hard to know with name "getcol"
+            return result.reshape(1, 1)
```

Comment:
> Should this be `reshape(-1, 1)` here?

If self is 1d, there will be only one value in the column requested, so in that sense `1` and `-1` will both work.

The comments above this line ask whether this should be a 2d single-value (1x1 matrix) in the 1d case. Does `getcol` imply it will be 2d? Or is it a replacement for indexing `self[:,j]` which would return a 1d object?

Reply:
We decided to raise for _getcol when called with a 1d array. Users should use indexing instead.